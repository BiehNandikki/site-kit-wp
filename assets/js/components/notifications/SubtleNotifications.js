/**
 * SubtleNotifications component.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * WordPress dependencies
 */
import { Fragment } from '@wordpress/element';

/**
 * Internal dependencies
 */
import useViewOnly from '../../hooks/useViewOnly';
import GA4AdSenseLinkedNotification from './GA4AdSenseLinkedNotification';
import { useAnalyticsAdSenseIntegrationCheck } from '../../hooks/useAnalyticsAdSenseIntegrationCheck';
import { useFeature } from '../../hooks/useFeature';

export default function SubtleNotifications() {
	const viewOnly = useViewOnly();

	const isGA4AdSenseIntegrationEnabled = useFeature(
		'ga4AdSenseIntegration'
	);
	const ga4AdSenseIntegration = useAnalyticsAdSenseIntegrationCheck();
	const renderGA4AdSenseLinkedNotification =
		ga4AdSenseIntegration.connected &&
		ga4AdSenseIntegration.linked &&
		isGA4AdSenseIntegrationEnabled;

	if ( viewOnly ) {
		return null;
	}

	return (
		<Fragment>
			{ renderGA4AdSenseLinkedNotification && (
				<GA4AdSenseLinkedNotification />
			) }
		</Fragment>
	);
}
