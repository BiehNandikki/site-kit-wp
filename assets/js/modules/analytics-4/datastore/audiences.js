/**
 * `modules/analytics-4` data store: audiences.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import API from 'googlesitekit-api';
import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';

const fetchGetAudiencesStore = createFetchStore( {
	baseName: 'getAudiences',
	controlCallback() {
		return API.get(
			'modules',
			'analytics-4',
			'audiences',
			{},
			{
				useCache: false,
			}
		);
	},
	reducerCallback( state, audiences ) {
		return { ...state, audiences };
	},
} );
fetchGetAudiencesStore();
