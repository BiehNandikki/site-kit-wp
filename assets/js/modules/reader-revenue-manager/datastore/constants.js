/**
 * `modules/reader-revenue-manager` data store constants.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

export const MODULES_READER_REVENUE_MANAGER = 'modules/reader-revenue-manager';

export const ERROR_CODE_NON_HTTPS_SITE = 'non_https_site';

export const ONBOARDING_STATE_UNSPECIFIED = 'ONBOARDING_STATE_UNSPECIFIED';

export const ONBOARDING_STATE_PENDING_VERIFICATION = 'PENDING_VERIFICATION';

export const ONBOARDING_STATE_ACTION_REQUIRED = 'ONBOARDING_ACTION_REQUIRED';

export const ONBOARDING_STATE_COMPLETE = 'ONBOARDING_COMPLETE';

export const UI_KEY_SHOW_RRM_PUBLICATION_APPROVED_NOTIFICATION =
	'show-rrm-publication-approved-notification';
