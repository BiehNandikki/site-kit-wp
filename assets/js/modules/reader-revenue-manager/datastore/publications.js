/**
 * `modules/reader-revenue-manager` data store: publications.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import API from 'googlesitekit-api';
import { commonActions, combineStores } from 'googlesitekit-data';
import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';
import {
	MODULES_READER_REVENUE_MANAGER,
	PUBLICATION_ONBOARDING_STATE,
} from './constants';

const fetchGetPublicationsStore = createFetchStore( {
	baseName: 'getPublications',
	controlCallback: () =>
		API.get(
			'modules',
			'reader-revenue-manager',
			'publications',
			{},
			{ useCache: true }
		),
	reducerCallback: ( state, publications ) => ( { ...state, publications } ),
} );

const baseInitialState = {
	publications: undefined,
};

const baseActions = {
	/**
	 * Finds a matched publication.
	 *
	 * @since n.e.x.t
	 *
	 * @return {Object|null} Matched publication; `null` if none found.
	 */
	*findMatchedPublication() {
		const { resolveSelect } = yield commonActions.getRegistry();
		const publications = yield commonActions.await(
			resolveSelect( MODULES_READER_REVENUE_MANAGER ).getPublications()
		);

		if ( publications.length === 0 ) {
			return null;
		}

		if ( publications.length === 1 ) {
			return publications[ 0 ];
		}

		const completedOnboardingPublication = publications.find(
			( publication ) =>
				publication.onboardingState ===
				PUBLICATION_ONBOARDING_STATE.ONBOARDING_COMPLETE
		);

		return completedOnboardingPublication || publications[ 0 ];
	},
};

const baseControls = {};

const baseReducer = ( state, { type } ) => {
	switch ( type ) {
		default:
			return state;
	}
};

const baseResolvers = {
	*getPublications() {
		const registry = yield commonActions.getRegistry();
		// Only fetch publications if there are none in the store.
		const publications = registry
			.select( MODULES_READER_REVENUE_MANAGER )
			.getPublications();
		if ( publications === undefined ) {
			yield fetchGetPublicationsStore.actions.fetchGetPublications();
		}
	},
};

const baseSelectors = {
	/**
	 * Gets list of publications associated with the account.
	 *
	 * @since n.e.x.t
	 *
	 * @param {Object} state Data store's state.
	 * @return {(Array.<Object>|undefined)} An array of publications; `undefined` if not loaded.
	 */
	getPublications( state ) {
		return state.publications;
	},
};

const store = combineStores( fetchGetPublicationsStore, {
	initialState: baseInitialState,
	actions: baseActions,
	controls: baseControls,
	reducer: baseReducer,
	resolvers: baseResolvers,
	selectors: baseSelectors,
} );

export const initialState = store.initialState;
export const actions = store.actions;
export const controls = store.controls;
export const reducer = store.reducer;
export const resolvers = store.resolvers;
export const selectors = store.selectors;

export default store;
