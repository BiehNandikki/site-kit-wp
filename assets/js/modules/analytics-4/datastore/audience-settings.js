/**
 * `modules/analytics-4` data store: audience settings.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
// import invariant from 'invariant';
// import { isPlainObject } from 'lodash';

/**
 * Internal dependencies
 */
import API from 'googlesitekit-api';
import Data from 'googlesitekit-data';
import { MODULES_ANALYTICS_4 } from './constants';
import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';
// import { createValidatedAction } from '../../../googlesitekit/data/utils';
import { createReducer } from '../../../googlesitekit/data/create-reducer';
// import { actions as errorStoreActions } from '../../../googlesitekit/data/create-error-store';

// const { receiveError, clearError } = errorStoreActions;

const fetchStoreReducerCallback = createReducer(
	( state, audienceSettings ) => {
		if ( ! state.audienceSettings ) {
			state.audienceSettings = {};
		}

		state.audienceSettings.settings = audienceSettings;
		state.audienceSettings.savedSettings = audienceSettings;
	}
);

const fetchGetAudienceSettingsStore = createFetchStore( {
	baseName: 'getAudienceSettings',
	controlCallback() {
		return API.get(
			'modules',
			'analytics-4',
			'audience-settings',
			{},
			{
				useCache: false,
			}
		);
	},
	reducerCallback: fetchStoreReducerCallback,
} );

const baseInitialState = {
	audienceSettings: undefined,
};

const baseActions = {};

const baseControls = {};

const baseReducer = ( state, { type } ) => {
	switch ( type ) {
		default: {
			return state;
		}
	}
};

const baseResolvers = {
	*getAudienceSettings() {
		const registry = yield Data.commonActions.getRegistry();

		const audienceSettings = registry
			.select( MODULES_ANALYTICS_4 )
			.getAudienceSettings();

		if ( audienceSettings === undefined ) {
			yield fetchGetAudienceSettingsStore.actions.fetchGetAudienceSettings();
		}
	},
};

const baseSelectors = {
	/**
	 * Gets the audience settings.
	 *
	 * @since n.e.x.t
	 *
	 * @param {Object} state Data store's state.
	 * @return {(Object|undefined)} Audience settings; `undefined` if not loaded.
	 */
	getAudienceSettings( state ) {
		return state.audienceSettings;
	},
};

const store = Data.combineStores( fetchGetAudienceSettingsStore, {
	initialState: baseInitialState,
	actions: baseActions,
	controls: baseControls,
	reducer: baseReducer,
	resolvers: baseResolvers,
	selectors: baseSelectors,
} );

export const initialState = store.initialState;
export const actions = store.actions;
export const controls = store.controls;
export const reducer = store.reducer;
export const resolvers = store.resolvers;
export const selectors = store.selectors;

export default store;
