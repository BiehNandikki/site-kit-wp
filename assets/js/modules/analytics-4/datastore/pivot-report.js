/**
 * `modules/analytics-4` data store: pivotReport.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import invariant from 'invariant';
import { isPlainObject } from 'lodash';

/**
 * Internal dependencies
 */
import API from 'googlesitekit-api';
import Data from 'googlesitekit-data';
import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';
import { MODULES_ANALYTICS_4 } from './constants';
import { stringifyObject } from '../../../util';
import { isValidDateRange } from '../../../util/report-validation';
import {
	normalizeReportOptions,
	isValidDimensionFilters,
	isValidDimensions,
	isValidMetrics,
	isValidMetricFilters,
	isValidPivots,
} from '../utils';

const fetchGetReportStore = createFetchStore( {
	baseName: 'getPivotReport',
	controlCallback: ( { options } ) => {
		return API.get(
			'modules',
			'analytics-4',
			'pivot-report',
			normalizeReportOptions( options )
		);
	},
	reducerCallback: ( state, report, { options } ) => {
		return {
			...state,
			pivotReports: {
				...state.pivotReports,
				[ stringifyObject( options ) ]: report,
			},
		};
	},
	argsToParams: ( options ) => {
		return { options };
	},
	validateParams: ( { options } = {} ) => {
		invariant(
			isPlainObject( options ),
			'options for Analytics 4 report must be an object.'
		);
		invariant(
			isValidDateRange( options ),
			'Either date range or start/end dates must be provided for Analytics 4 report.'
		);

		const {
			metrics,
			dimensions,
			dimensionFilters,
			metricFilters,
			pivots,
			orderby,
			limit,
		} = normalizeReportOptions( options );

		invariant(
			metrics.length,
			'Requests must specify at least one metric for an Analytics 4 report.'
		);
		invariant(
			isValidMetrics( metrics ),
			'metrics for an Analytics 4 report must be either a string, an array of strings, an object, an array of objects, or a mix of strings and objects. Objects must have a "name" property. Metric names must match the expression ^[a-zA-Z0-9_]+$.'
		);
		invariant(
			isValidPivots( pivots ),
			'pivots for an Analytics 4 report must be an array of objects. Each object must have a "fieldNames" property and a "limit".'
		);

		if ( orderby ) {
			invariant(
				Array.isArray( orderby ),
				'orderby for an Analytics 4 pivot report must be passed within a pivot.'
			);
		}
		if ( limit ) {
			invariant(
				typeof limit === 'number',
				'limit for an Analytics 4 pivot report must be passed within a pivot.'
			);
		}

		if ( dimensions ) {
			invariant(
				isValidDimensions( dimensions ),
				'dimensions for an Analytics 4 report must be either a string, an array of strings, an object, an array of objects, or a mix of strings and objects. Objects must have a "name" property.'
			);
		}

		if ( dimensionFilters ) {
			invariant(
				isValidDimensionFilters( dimensionFilters ),
				'dimensionFilters for an Analytics 4 report must be a map of dimension names as keys and dimension values as values.'
			);
		}

		if ( metricFilters ) {
			invariant(
				isValidMetricFilters( metricFilters ),
				'metricFilters for an Analytics 4 report must be a map of metric names as keys and filter value(s) as numeric fields, depending on the filterType.'
			);
		}
	},
} );

const baseInitialState = {
	pivotReports: {},
};

const baseResolvers = {
	*getPivotReport( options = {} ) {
		const registry = yield Data.commonActions.getRegistry();
		const existingReport = registry
			.select( MODULES_ANALYTICS_4 )
			.getPivotReport( options );

		// If there is already a report loaded in state, consider it fulfilled
		// and don't make an API request.
		if ( existingReport ) {
			return;
		}

		yield fetchGetReportStore.actions.fetchGetPivotReport( options );
	},
};

const baseSelectors = {
	/**
	 * Gets an Analytics pivot report for the given options.
	 *
	 * @since n.e.x.t
	 *
	 * @param {Object}         state                      Data store's state.
	 * @param {Object}         options                    Options for generating the report.
	 * @param {string}         options.startDate          Required. Start date to query report data for as YYYY-mm-dd.
	 * @param {string}         options.endDate            Required. End date to query report data for as YYYY-mm-dd.
	 * @param {Array.<string>} options.metrics            Required. List of metrics to query.
	 * @param {Array.<string>} options.dimensions         Required. List of dimensions to group results by.
	 * @param {Array.<string>} options.pivots             Required. List of pivots to structure the report data into columns.
	 * @param {Object}         [options.dimensionFilters] Optional. Map of dimension filters for filtering options on a dimension. Default an empty object.
	 * @param {Object}         [options.metricFilters]    Optional. Map of metric filters for filtering options on a metric. Default an empty object.
	 * @param {string}         [options.url]              Optional. URL to get a report for only this URL. Default an empty string.
	 * @return {(Array.<Object>|undefined)} An Analytics report; `undefined` if not loaded.
	 */
	getPivotReport( state, options ) {
		const { pivotReports } = state;

		return pivotReports[ stringifyObject( options ) ];
	},
};

const store = Data.combineStores( fetchGetReportStore, {
	initialState: baseInitialState,
	resolvers: baseResolvers,
	selectors: baseSelectors,
} );

export const initialState = store.initialState;
export const actions = store.actions;
export const controls = store.controls;
export const reducer = store.reducer;
export const resolvers = store.resolvers;
export const selectors = store.selectors;

export default store;
