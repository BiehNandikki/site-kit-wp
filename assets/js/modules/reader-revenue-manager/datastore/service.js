/**
 * `modules/reader-revenue-manager` data store.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * External dependencies
 */
import invariant from 'invariant';

/**
 * WordPress dependencies
 */
import { addQueryArgs } from '@wordpress/url';

/**
 * Internal dependencies
 */
import { createRegistrySelector } from 'googlesitekit-data';
import { CORE_USER } from '../../../googlesitekit/datastore/user/constants';

/**
 * Link to RRM platform.
 */
const publicationCenterURL = 'https://publishercenter.google.com';

const selectors = {
	/**
	 * Returns a link to the Reader Revenue Manager platform.
	 *
	 * @since n.e.x.t
	 *
	 * @param {string} publicationID Publication ID.
	 * @return {string} Link to Reader Revenue Management platform.
	 */
	getServiceURL: createRegistrySelector(
		( select ) =>
			( state, publicationID = null ) => {
				// If there is no publication ID, return link to platform.
				if ( ! publicationID ) {
					return publicationCenterURL;
				}

				invariant(
					'string' === typeof publicationID,
					'Publication ID must be string.'
				);

				const linkRRM = addQueryArgs(
					publicationCenterURL + '/reader-revenue-manager',
					{
						publication: publicationID,
						utm_source: 'sitekit',
					}
				);

				const accountChooserBaseURI =
					select( CORE_USER ).getAccountChooserURL( linkRRM );

				if ( accountChooserBaseURI === undefined ) {
					return undefined;
				}

				return accountChooserBaseURI;
			}
	),
};

const store = {
	selectors,
};

export default store;
