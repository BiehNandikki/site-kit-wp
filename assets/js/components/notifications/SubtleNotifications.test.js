/**
 * SubtleNotifications component tests.
 *
 * Site Kit by Google, Copyright 2024 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/**
 * Internal dependencies
 */
import SubtleNotifications from './SubtleNotifications';
import {
	render,
	createTestRegistry,
	provideModules,
} from '../../../../tests/js/test-utils';
import { CORE_USER } from '../../googlesitekit/datastore/user/constants';
import { MODULES_ANALYTICS_4 } from '../../modules/analytics-4/datastore/constants';

describe( 'GA4AdSenseLinkedNotification', () => {
	let registry;

	const analyticsReport = new RegExp(
		'^/google-site-kit/v1/modules/analytics-4/data/report'
	);

	beforeEach( () => {
		registry = createTestRegistry();
		provideModules( registry );
	} );

	it( 'does not render GA4AdSenseLinkedNotification if AdSense module is not active', async () => {
		registry.dispatch( MODULES_ANALYTICS_4 ).setSettings( {
			adSenseLinked: true,
		} );
		provideModules( registry, [
			{
				active: true,
				connected: true,
				slug: 'analytics-4',
			},
			{
				active: false,
				connected: false,
				slug: 'adsense',
			},
		] );

		const { container, waitForRegistry } = render(
			<SubtleNotifications />,
			{
				registry,
				features: [ 'ga4AdSenseIntegration' ],
			}
		);
		await waitForRegistry();

		expect( container.childElementCount ).toBe( 0 );
	} );

	it( 'does not render GA4AdSenseLinkedNotification if AdSense and Analytics are not linked', async () => {
		registry.dispatch( MODULES_ANALYTICS_4 ).setSettings( {
			adSenseLinked: false,
		} );
		provideModules( registry, [
			{
				active: true,
				connected: true,
				slug: 'analytics-4',
			},
			{
				active: true,
				connected: true,
				slug: 'adsense',
			},
		] );

		const { container, waitForRegistry } = render(
			<SubtleNotifications />,
			{
				registry,
				features: [ 'ga4AdSenseIntegration' ],
			}
		);
		await waitForRegistry();

		expect( container.childElementCount ).toBe( 0 );
	} );

	it( 'renders GA4AdSenseLinkedNotification when both Analytics and AdSense modules are active', async () => {
		fetchMock.getOnce( analyticsReport, {
			body: {
				rowCount: null,
			},
			status: 200,
		} );

		provideModules( registry, [
			{
				active: true,
				connected: true,
				slug: 'analytics-4',
			},
			{
				active: true,
				connected: true,
				slug: 'adsense',
			},
		] );

		registry.dispatch( MODULES_ANALYTICS_4 ).setSettings( {
			adSenseLinked: true,
		} );

		registry.dispatch( CORE_USER ).receiveGetDismissedItems( [] );

		const { container, waitForRegistry } = render(
			<SubtleNotifications />,
			{
				registry,
				features: [ 'ga4AdSenseIntegration' ],
			}
		);
		await waitForRegistry();

		expect( container ).toHaveTextContent(
			'Your AdSense and Analytics accounts are linked'
		);
		expect( container ).toMatchSnapshot();
	} );
} );
