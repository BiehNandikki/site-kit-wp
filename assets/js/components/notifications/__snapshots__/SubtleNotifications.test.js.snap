// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GA4AdSenseLinkedNotification renders GA4AdSenseLinkedNotification when both Analytics and AdSense modules are active 1`] = `
<div>
  <div
    class="mdc-layout-grid"
  >
    <div
      class="mdc-layout-grid__inner"
    >
      <div
        class="googlesitekit-dashboard-notifications mdc-layout-grid__cell mdc-layout-grid__cell--align-middle mdc-layout-grid__cell--span-12"
      >
        <div
          class="googlesitekit-dashboard-notifications__icon"
        >
          <svg />
        </div>
        <div
          class="googlesitekit-dashboard-notifications__content"
        >
          <div>
            Your AdSense and Analytics accounts are linked.
          </div>
          <div>
            We’ll let you know as soon as there’s enough data available.
          </div>
        </div>
        <button
          class="mdc-button mdc-button--tertiary"
          target="_self"
        >
          <span
            class="mdc-button__label"
          >
            Got it
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
